File: packages/optio-agents/src/optio_agents/agent_info.py
from dataclasses import dataclass


@dataclass(frozen=True)
class AgentInfo:
    """Canonical, user-facing metadata for an agent engine."""

    slug: str  # machine id, equals the engine's agent_type ("claudecode")
    name: str  # canonical user-facing name ("Claude Code")
    url: str   # canonical product URL

File: packages/optio-agents-ui/src/agent-info.ts
import { AGENTS, type AgentType } from './agents.generated.js';

/** Canonical, user-facing metadata for an agent engine. */
export interface AgentInfo {
  slug: AgentType;
  name: string;
  url: string;
}

/** Canonical metadata for the given engine slug. */
export function getAgentInfo(slug: AgentType): AgentInfo {
  return AGENTS[slug];
}
